test/view/iota.cpp

```diff
@@ -214,6 +214,15 @@ int main()
         CHECK((i - cstr2.begin()) == 4);
     }
 
+
+    {
+       // test step
+       ::check_equal(views::iota(0, 10, 2), {0, 2, 4, 6, 8});
+       ::check_equal(views::iota(5, 0, -1), {5, 4, 3, 2, 1});
+       ::check_equal(size(views::iota(1, 9, 3)), 3U);
```

Comment:
If you change this to:

```cpp
::check_equal(views::iota(1, 9, 3), {1, 4, 7});
```

The test will fail, because `iota(1, 9, 3)` at the moment is an infinite stream of ints... despite having a correct fixed `size`.

Reply:
Seems like `cursor` is missing an `equal` overload that `requires detail::iota_stridable_<From, To>` equivalent to the `sentinel` overload.

Reply:
Hum indeed - I will have to fix that. Might take me a few days to get to it, I need to do something a bit more clever than in the sentinel case

Reply:
cursors don't necessary represent the end - I think what i need to do is introduce a `done_` member as is done for `closed_iota_view` - such that cursors compare equals if both are done or have the same `from_`

Reply:
It works in the general case, so you can compare any two cursors from the same `iota_view`. If you use any cursor in its range as the end, and iterate from the begin, you can be sure that it'll eventually reach that end. That makes me think, is there a test for reverse iterating `iota_view`s?